database/src/main/postgres/flows/V1.7.5__get_flow_checkpoints.sql

```diff
@@ -0,0 +1,127 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE FUNCTION flows.get_flow_checkpoints(
+    IN i_partitioning_of_flow JSONB,
+    IN i_limit                INT DEFAULT 5,
+    IN i_checkpoint_name      TEXT DEFAULT NULL,
+    OUT status                INTEGER,
+    OUT status_text           TEXT,
+    OUT id_checkpoint         UUID,
+    OUT checkpoint_name       TEXT,
+    OUT measure_name          TEXT,
+    OUT measure_columns       TEXT[],
+    OUT measurement_value     JSONB,
+    OUT checkpoint_start_time TIMESTAMP WITH TIME ZONE,
+    OUT checkpoint_end_time   TIMESTAMP WITH TIME ZONE
+) RETURNS SETOF record AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: flows.get_flow_checkpoints(3)
+--      Retrieves all checkpoints (measures and their measurement details) related to a primary flow
+--      associated with the input partitioning.
+--
+-- Note: a single row returned from this function doesn't contain all data related to a single checkpoint - it only
+--     represents one measure associated with a checkpoint. So even if only a single checkpoint would be retrieved,
+--     this function can potentially return multiple rows.
+--
+-- Note: checkpoints will be retrieved in ordered fashion, by checkpoint_time and id_checkpoint
+--
+-- Parameters:
+--      i_partitioning_of_flow  - partitioning to use for identifying the flow associate with checkpoints
+--                                that will be retrieved
+--      i_limit                 - (optional) maximum number of checkpoints retrieved
+--                                if zero or lower specified, all data will be returned, i.e. no limit will be applied
+--      i_checkpoint_name       - (optional) if specified, returns data related to particular checkpoint's name
+--
+--      Note: checkpoint name uniqueness is not enforced by the data model, so there can be multiple different
+--          checkpoints with the same name in the DB, i.e. multiple checkpoints can be retrieved even when
+--          specifying `i_checkpoint_name` parameter
+--
+-- Returns:
+--      status              - Status code
+--      status_text         - Status text
+--      id_checkpoint       - id of retrieved checkpoint
+--      checkpoint_name     - name of retrieved checkpoint
+--      measure_name        - measure name associated with a given checkpoint
+--      measure_columns     - measure columns associated with a given checkpoint
+--      measurement_value   - measurement details associated with a given checkpoint
+--      checkpoint_time     - time
+--
+-- Status codes:
+--      11                  - Flow created
+--      16                  - No checkpoints found
```

Comment:
Would not use that status code imho. No checkpoints -> empty result set.

Reply:
There is (as usual) a reasoning behind this:
* consistency - get_partitioning_additional_data and get_partitioning_measures for example have them
* it doesn't hurt to have the end user more informed, it's just a few bytes on the network traffic

Reply:
It's being removed actually in #167  😉

Reply:
but still, it's extremely cheap once we have https://github.com/AbsaOSS/fa-db/pull/120 (what was the point of implementing it then?)

Reply:
I was thinking about it. Really do remove the status 16. It would complicate the Scala code. It would handle the case when there's an OK status, but still cannot (and shouldn't) create the output class (all `NULL`).

Reply:
Agreed